client/src/main/java/org/hyperskill/database/clientside/Client.java

```diff
@@ -1,34 +1,65 @@
 package org.hyperskill.database.clientside;
 
-import java.io.*;
-import java.net.*;
-import java.util.*;
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.net.Socket;
 
 public class Client {
-  private static final String SERVER_ADDRESS = "127.0.0.1";
-  private static final int SERVER_PORT = 34522;
-
-  public static void main(String[] args) {
-      try (
-          Socket socket = new Socket(SERVER_ADDRESS, SERVER_PORT);
-          DataInputStream input = new DataInputStream(socket.getInputStream());
-          DataOutputStream output  = new DataOutputStream(socket.getOutputStream())
-      ) {
-          System.out.println("Client started!");
-          while (true) {
-            System.out.print("Give me a record # ");
-            Scanner scanner = new Scanner(System.in);
-            String msg = scanner.nextLine();
-
-                output.writeUTF(msg); // sending message to the server
-                System.out.println("Sent: " + msg);
-                String receivedMsg = input.readUTF(); // response message
-                System.out.println("Recieved: " + receivedMsg);
-
-
-          }
-      } catch (IOException e) {
-          e.printStackTrace();
-      }
-  }
+    @Parameter(names = {"-host", "-ip"}, description = "IP address")
+    String host;
+
+    @Parameter(names = {"-port"}, description = "Port", validateWith = PositiveInteger.class)
+    int port;
+
+    @Parameter(names = "-t", description = "Command", validateWith = Commands.class)
+    String command;
+
+    @Parameter(names = "-m", description = "Text to save")
+    String text;
+
+    @Parameter(names = "-i", description = "Index", validateWith = PositiveInteger.class)
+    int index;
+
+    public static void main(String... argv) {
+        Client client = new Client();
+        JCommander.newBuilder().addObject(client).build().parse(argv);
+
+        if (argv.length < 8) {
+            System.err.println(
+                    "Usage: java -jar client.jar -ip <hostname> -port <port> -t <get/set/delete> -i <index> [-m <message>]");
+            System.exit(1);
+        }
+
+        try (
+                Socket socket = new Socket(client.host, client.port);
+                DataInputStream input = new DataInputStream(socket.getInputStream());
+                DataOutputStream output = new DataOutputStream(socket.getOutputStream())
+        ) {
+            if (client.command.equals("set")) {
```

Comment:
In `Commands` validator you use `value.trim().toLowerCase()`, but here you don't preprocess `command`, can lead to a potential bug

Also, when checking strings against hardcoded values, better to switch their places, e.g. `"set".equals(client.command)`, or use `Objects.equals(client.command, "set")`, to avoid NullPointerException
Another approach would be to use `switch`, e.g. `switch (client.command.trim().toLowerCase()) { case "get": ... }`